### PR TITLE
use existing secret for  keycloakx database password

### PR DESCRIPTION
**Pull request by ismailyenigul**
This path allows us to use existing k8s secret to set database password instead of setting it in values.yml plain text.
This approach is good for GitOps not to keep secret in plaintext in git repo.
the secret can be created externally and save t encrypted(eg sealed-secrets) in git repo.
PS: I will push another commit for  README.md if you update it manually.

### CODE REVIEW

File: charts/keycloakx/templates/_helpers.tpl [hunk 1]
```diff
@@ -64,3 +64,17 @@ Create the service DNS name.
...
 {{ include "keycloak.fullname" . }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
 {{- end }}
+
+{{- define "keycloak.databasePasswordEnv" -}}
+{{- if or .Values.database.password .Values.database.existingSecret -}}
+- name: KC_DB_PASSWORD
+  valueFrom:
+    secretKeyRef:
+    {{- if .Values.database.existingSecret }}
+      name: {{ .Values.database.existingSecret }}
```
Comment (grieshaber) [inline charts/keycloakx/templates/_helpers.tpl:74]:
```suggestion
      name: {{ .Values.database.existingSecret | default (printf "%s-database" (include "keycloak.fullname" . ))}}
```
and drop the rest of the if clause below.

File: charts/keycloakx/values.yaml [hunk 2]
```diff
@@ -364,6 +364,8 @@ dbchecker:
       memory: "32Mi"
 
 database:
+  # don't create secret for db password. Instead use existing k8s secret
+  # existingSecret: "my-existent-dbpass-secret"
```
Comment (grieshaber) [inline charts/keycloakx/values.yaml:368]:
```suggestion
  # existingSecret: "my-existent-dbpass-secret"
  existingSecret: ""
```
to avoid nil pointer

**Review by grieshaber — CHANGES_REQUESTED** [review]
Thanks for your PR - i made some suggestions with small "improvements".

Regarding all the release relevant stuff - as mentioned at the changes, please revert them, as i will take care of an cumulated release later on.

**Comment by grieshaber** [comment]
@ismailyenigul could also please make sure, that your commits contain a valid signature? This is needed to check the DCO action

**Comment by ismailyenigul** [comment]
sorry for new releases. i had to test it in my env first. now going to apply changes.

**Comment by grieshaber** [comment]
no worries, in general it is fine to do so. But, as we just need to catch up our maintenance of this repo, things are a bit "more manual" than needed.

Thanks for the fast updates of this PR! I see you resolved the discussion and added a commit message with "added suggested changes" - but, if i'm not mistaken, the commit did not contain the suggested changes.

**Comment by ismailyenigul** [comment]
just pushed new commit.
Please check

**Review by grieshaber — APPROVED** [review]

**Pull request merged**